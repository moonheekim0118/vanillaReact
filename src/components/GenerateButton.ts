interface Props {
    title: string;
    onClick: () => Promise<void>;
}

// 버튼 클릭시 image와 quotes 모두 갱신
const GenerateButton = () => {
    function render({ title, onClick }: Props): Element {
        const Button = document.createElement("button");
        Button.className = "Generate-Btn";
        Button.innerText = title;
        Button.addEventListener("click", onClick);

        return Button;
    }
    return render;
};

export default GenerateButton();
