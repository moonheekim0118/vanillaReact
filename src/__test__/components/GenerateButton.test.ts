import "jsdom-global/register";
import GenerateButton from "../../components/GenerateButton";

describe("<GenerateButton/>", () => {
    let container = null;
    const mockFn = jest.fn();

    it("renders correctly", () => {
        container = GenerateButton({ title: "test", onClick: mockFn });
    });

    it("matches snapshot", () => {
        expect(container).toMatchSnapshot();
    });

    it("should call onClick if button is clicked", () => {
        container.click();
        expect(mockFn).toHaveBeenCalled();
    });
});
