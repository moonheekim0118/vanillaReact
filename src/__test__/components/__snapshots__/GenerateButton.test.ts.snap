// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenerateButton/> matches snapshot 1`] = `
<button
  class="Generate-Btn"
/>
`;
